src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -291,6 +335,27 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                             .show();
                     break;
             }
+        } else if (requestCode == PICK_BACKUP) {
+            Uri uri = data.getData();
```

Comment:
```suggestion
            Uri uri = (data != null ? data.getData() : null);
```

there is a `NullPointerException` here if user press back instead of picking a file.

Reply:
good catch!